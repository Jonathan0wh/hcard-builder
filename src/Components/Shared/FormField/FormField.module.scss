@import 'styles/colors';

.form_field_col {
  :global {
    .ant-form-item {
      padding-bottom: 0;
      margin-bottom: 16px;
    }
    .ant-form-item-label {
      padding: 0;
    }
    label {
      font-size: 12px;
      font-weight: 400;
      color: $bluewood;
    }
    label::after {
      content: none;
    }
    input[type='text'],
    input[type='email'],
    input[type='tel'] {
      font-family: 'Merriweather';
      font-size: 16px;
      font-weight: 400;
      color: $black;
      height: 36px;
    }
  }
}
