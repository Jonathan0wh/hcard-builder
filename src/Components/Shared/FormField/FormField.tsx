import React from 'react';
import { Col, Form, Input } from 'antd';
import v from 'voca';

import styles from './FormField.module.scss';

interface FormFieldProps {
  label: string;
  value: string;
  setValue: (newValue: string) => void;
  type?: string;
}

const FormField = ({
  label,
  value,
  setValue,
  type = 'text'
}: FormFieldProps) => (
  <Col lg={12} className={styles.form_field_col}>
    <Form.Item label={label} htmlFor={v.kebabCase(label)}>
      <Input
        id={v.kebabCase(label)}
        name={v.kebabCase(label)}
        value={value}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
          setValue(e.target.value)
        }
        type={type}
      />
    </Form.Item>
  </Col>
);

export default FormField;
