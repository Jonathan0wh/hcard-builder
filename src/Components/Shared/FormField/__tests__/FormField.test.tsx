import React from 'react';
import { shallow } from 'enzyme';

import FormField from '../FormField';

describe('<FormField />', () => {
  it('should render by default', () => {
    const wrapper = shallow(
      <FormField label="Text" value={''} setValue={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render email field', () => {
    const wrapper = shallow(
      <FormField label="Text" value={''} setValue={jest.fn()} type="email" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render tel input', () => {
    const wrapper = shallow(
      <FormField label="Text" value={''} setValue={jest.fn()} type="tel" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
