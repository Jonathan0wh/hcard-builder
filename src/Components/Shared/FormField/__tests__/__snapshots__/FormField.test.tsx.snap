// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField /> should render by default 1`] = `
<Col
  className="form_field_col"
  lg={12}
>
  <FormItem
    hasFeedback={false}
    htmlFor="text"
    label="Text"
  >
    <Input
      id="text"
      name="text"
      onChange={[Function]}
      type="text"
      value=""
    />
  </FormItem>
</Col>
`;

exports[`<FormField /> should render email field 1`] = `
<Col
  className="form_field_col"
  lg={12}
>
  <FormItem
    hasFeedback={false}
    htmlFor="text"
    label="Text"
  >
    <Input
      id="text"
      name="text"
      onChange={[Function]}
      type="email"
      value=""
    />
  </FormItem>
</Col>
`;

exports[`<FormField /> should render tel input 1`] = `
<Col
  className="form_field_col"
  lg={12}
>
  <FormItem
    hasFeedback={false}
    htmlFor="text"
    label="Text"
  >
    <Input
      id="text"
      name="text"
      onChange={[Function]}
      type="tel"
      value=""
    />
  </FormItem>
</Col>
`;
